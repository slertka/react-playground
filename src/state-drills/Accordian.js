import React from "react";

class Accordian extends React.Component {
  static defaultProps = {
    sections: []
  };

  state = {
    activeTabIndex: null
  };

  displayContent(index) {
    this.setState({
      activeTabIndex: index
    });
  }

  renderAccordian() {
    return this.props.sections.map((tab, index) => (
      <li>
        <button key={index} onClick={() => this.displayContent(index)}>
          {tab.title}
        </button>
        <p>{index === this.state.activeTabIndex ? tab.content : ""}</p>
      </li>
    ));
  }

  render() {
    return (
      <div>
        <ul>{this.renderAccordian()}</ul>
      </div>
    );
  }
}

export default Accordian;
