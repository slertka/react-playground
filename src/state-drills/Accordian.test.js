import React from "react";
import ReactDOM from "react-dom";
import { shallow } from "enzyme";
import toJson from "enzyme-to-json";

import Accordian from "./Accordian";

describe("Accordian component", () => {
  const sectionsArray = [
    {
      title: "Id minim dolor sit deserunt do sint dolore.",
      content:
        "Dolore enim do amet dolore nisi nulla magna do. Dolor voluptate ullamco ut ea eiusmod fugiat enim culpa amet. Culpa consequat velit ipsum amet minim consequat dolore cillum cupidatat adipisicing cillum exercitation ut. Sit consequat ex elit incididunt amet culpa dolor Lorem mollit eiusmod non est aliqua officia."
    },
    {
      title:
        "Esse cupidatat ut deserunt eiusmod eu minim voluptate laboris voluptate ad sit do cillum.",
      content:
        "Qui qui esse labore quis adipisicing est dolore velit nostrud magna dolor. Deserunt nisi nulla ullamco nulla exercitation. Enim nostrud velit quis officia incididunt reprehenderit do magna officia deserunt. Ipsum duis fugiat mollit Lorem consectetur consequat ut cupidatat minim est quis enim proident. Ut incididunt enim qui incididunt. Velit exercitation nisi ea exercitation labore ipsum sunt proident ex magna officia occaecat elit mollit. Sunt in aliqua duis cillum fugiat cupidatat est anim ad commodo consectetur nisi."
    },
    {
      title:
        "Fugiat ut dolor culpa aute consectetur laborum reprehenderit velit labore pariatur consectetur ex consectetur.",
      content:
        "Nulla reprehenderit nulla ullamco officia. Culpa do voluptate ipsum amet aliqua eiusmod ipsum commodo. Ex ut pariatur reprehenderit non est elit excepteur ad pariatur dolore aliquip commodo proident. Sint sit do velit sit nisi. Duis aliqua aliqua duis laborum deserunt elit eiusmod enim eiusmod nulla minim sit magna."
    },
    {
      title:
        "Pariatur ad laboris cillum consequat labore sint fugiat nostrud id in proident esse.",
      content:
        "Duis proident irure do quis reprehenderit Lorem esse adipisicing magna esse sint exercitation mollit anim. Proident incididunt eu eiusmod dolor ut labore ea voluptate ut laborum ad minim veniam in. Laborum enim commodo sit eiusmod quis sunt minim. Consectetur et dolor culpa non cillum enim dolore velit proident velit ut consequat consequat proident. Ut consequat Lorem laboris velit exercitation culpa elit Lorem nulla enim reprehenderit."
    }
  ];

  it("Renders without errors", () => {
    const div = document.createElement("div");
    ReactDOM.render(<Accordian />, div);
    ReactDOM.unmountComponentAtNode(div);
  });

  it("Renders empty when sections props is not provided", () => {
    const wrapper = shallow(<Accordian />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it("Does not render any content by default", () => {
    const wrapper = shallow(<Accordian sections={sectionsArray} />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it("Displays content in the respective section in the list when its button is clicked", () => {
    const wrapper = shallow(<Accordian sections={sectionsArray} />);
    wrapper
      .find("button")
      .at(2)
      .simulate("click");
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it("Only displays content of the last section clicked when multiple are clicked", () => {
    const wrapper = shallow(<Accordian sections={sectionsArray} />);
    wrapper
      .find("button")
      .at(2)
      .simulate("click");
    wrapper
      .find("button")
      .at(3)
      .simulate("click");
    wrapper
      .find("button")
      .at(0)
      .simulate("click");
    wrapper
      .find("button")
      .at(1)
      .simulate("click");
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
