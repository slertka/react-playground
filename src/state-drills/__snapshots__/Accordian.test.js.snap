// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Accordian component Displays content in the respective section in the list when its button is clicked 1`] = `
<div>
  <ul>
    <li>
      <button
        key="0"
        onClick={[Function]}
      >
        Id minim dolor sit deserunt do sint dolore.
      </button>
      <p />
    </li>
    <li>
      <button
        key="1"
        onClick={[Function]}
      >
        Esse cupidatat ut deserunt eiusmod eu minim voluptate laboris voluptate ad sit do cillum.
      </button>
      <p />
    </li>
    <li>
      <button
        key="2"
        onClick={[Function]}
      >
        Fugiat ut dolor culpa aute consectetur laborum reprehenderit velit labore pariatur consectetur ex consectetur.
      </button>
      <p>
        Nulla reprehenderit nulla ullamco officia. Culpa do voluptate ipsum amet aliqua eiusmod ipsum commodo. Ex ut pariatur reprehenderit non est elit excepteur ad pariatur dolore aliquip commodo proident. Sint sit do velit sit nisi. Duis aliqua aliqua duis laborum deserunt elit eiusmod enim eiusmod nulla minim sit magna.
      </p>
    </li>
    <li>
      <button
        key="3"
        onClick={[Function]}
      >
        Pariatur ad laboris cillum consequat labore sint fugiat nostrud id in proident esse.
      </button>
      <p />
    </li>
  </ul>
</div>
`;

exports[`Accordian component Does not render any content by default 1`] = `
<div>
  <ul>
    <li>
      <button
        key="0"
        onClick={[Function]}
      >
        Id minim dolor sit deserunt do sint dolore.
      </button>
      <p />
    </li>
    <li>
      <button
        key="1"
        onClick={[Function]}
      >
        Esse cupidatat ut deserunt eiusmod eu minim voluptate laboris voluptate ad sit do cillum.
      </button>
      <p />
    </li>
    <li>
      <button
        key="2"
        onClick={[Function]}
      >
        Fugiat ut dolor culpa aute consectetur laborum reprehenderit velit labore pariatur consectetur ex consectetur.
      </button>
      <p />
    </li>
    <li>
      <button
        key="3"
        onClick={[Function]}
      >
        Pariatur ad laboris cillum consequat labore sint fugiat nostrud id in proident esse.
      </button>
      <p />
    </li>
  </ul>
</div>
`;

exports[`Accordian component Only displays content of the last section clicked when multiple are clicked 1`] = `
<div>
  <ul>
    <li>
      <button
        key="0"
        onClick={[Function]}
      >
        Id minim dolor sit deserunt do sint dolore.
      </button>
      <p />
    </li>
    <li>
      <button
        key="1"
        onClick={[Function]}
      >
        Esse cupidatat ut deserunt eiusmod eu minim voluptate laboris voluptate ad sit do cillum.
      </button>
      <p>
        Qui qui esse labore quis adipisicing est dolore velit nostrud magna dolor. Deserunt nisi nulla ullamco nulla exercitation. Enim nostrud velit quis officia incididunt reprehenderit do magna officia deserunt. Ipsum duis fugiat mollit Lorem consectetur consequat ut cupidatat minim est quis enim proident. Ut incididunt enim qui incididunt. Velit exercitation nisi ea exercitation labore ipsum sunt proident ex magna officia occaecat elit mollit. Sunt in aliqua duis cillum fugiat cupidatat est anim ad commodo consectetur nisi.
      </p>
    </li>
    <li>
      <button
        key="2"
        onClick={[Function]}
      >
        Fugiat ut dolor culpa aute consectetur laborum reprehenderit velit labore pariatur consectetur ex consectetur.
      </button>
      <p />
    </li>
    <li>
      <button
        key="3"
        onClick={[Function]}
      >
        Pariatur ad laboris cillum consequat labore sint fugiat nostrud id in proident esse.
      </button>
      <p />
    </li>
  </ul>
</div>
`;

exports[`Accordian component Renders empty when sections props is not provided 1`] = `
<div>
  <ul />
</div>
`;
